counterparty-core/counterpartycore/lib/messages/utxo.py

```diff
@@ -60,19 +87,13 @@ def validate(db, source, destination, asset, quantity, block_index=None):
     else:
         fee = 0
 
-    # check if source has enough funds
-    asset_balance = ledger.get_balance(db, source, asset)
-    if asset == config.XCP:
-        # fee is always paid in XCP
-        if asset_balance < quantity + fee:
-            problems.append("insufficient funds for transfer and fee")
-    else:
-        if asset_balance < quantity:
-            problems.append("insufficient funds for transfer")
-        if source_is_address:
-            xcp_balance = ledger.get_balance(db, source, config.XCP)
-            if xcp_balance < fee:
-                problems.append("insufficient funds for fee")
+    if source_is_address and util.enabled("spend_utxo_to_detach"):
```

Comment:
Can we refactor this and the next if block to make the logic more clear and avoid the early return?

Reply:
well seen, done!